src/data/finset/basic.lean

```diff
@@ -2065,6 +2161,10 @@ theorem map_disj_union_aux {f : α ↪ β} {s₁ s₂ : finset α} :
   (∀ a, a ∈ s₁ → a ∉ s₂) ↔ ∀ a, a ∈ map f s₁ → a ∉ map f s₂ :=
 by simp_rw [forall_mem_map, mem_map']
 
+@[simp] lemma disjoint_map [decidable_eq α] [decidable_eq β] {s t : finset α} (f : α ↪ β) :
+  disjoint (s.map f) (t.map f) ↔ disjoint s t :=
+by rw [disjoint_left, disjoint_left, ←map_disj_union_aux]
```

Comment:
This lemma along with `disjoint_image` are the main golfs from this change; but the longer term goal is to unify the hypothesis taken by `finset.disj_union` with `disjoint`.